doc/ug/rust_for_c.md

```diff
@@ -0,0 +1,2310 @@
+
+[^22]: It has some issues, similar to the C variant.
+See [https://doc.rust-lang.org/nomicon/other-reprs.html#reprpacked](https://doc.rust-lang.org/nomicon/other-reprs.html#reprpacked)
+
+[^23]: Unlike C, Rust has true zero-sized types (ZSTs).
+They are completely elided from the program at runtime, but can be useful for creating abstractions.
+
+[^24]: This is similar to how a `bool` must always be `0` or `1`.
+One could even imagine that `bool` is defined as `enum bool { false = 0, true = 1 }`.
+
+[^25]: Without uttering `unsafe`, it is impossible to witness uninitialized or invalid data.
+Doing so even with `unsafe` is Undefined Behavior.
+
+[^26]: This feature is also sometimes known as "algebraic data types".
+
+[^27]: This syntax requires that the array component type be "copyable", which we will get to later.
+
+[^28]: These accesses are often elided.
+LLVM is very good at optimizing them away, but not perfect.
+
+[^29]: In C, pointers must _always_ be well-aligned.
+Rust only requires this when dereferencing the pointer.
+
+[^30]: In other words, Rust's aliasing model for pointers is the same as that of `-fno-strict-aliasing`.
+Rust allows `*mut u32` and `*mut u64`to alias, for example.
+
+[^31]: In C, it is technically UB to forge raw pointers (such as creating a pointer to an MMIO register) and then dereference them, although all embedded programs need to do this anyway.
+While Rust has not yet formalized what this means, it seems unlikely that they will make forging raw pointers to MMIO regions UB.
+
+[^32]: Or by casting zero.
+
+[^33]: Rust currently does not have an equivalent of the `->` operator, though it may get one some day.
+As such, it is not possible to get pointers to fields of a pointer-to-struct without creating a reference along the way.
+
+[^34]: We will get into references later.
+
+[^35]: We will learn more about "move semantics" when we discuss ownership.
+
+[^36]: However, great care should be taken when using these methods on types that track ownership of a resource, since this can result in a double-free when the pointee is freed.
+
+[^37]: [https://doc.rust-lang.org/std/primitive.pointer.html#method.read_unaligned](https://doc.rust-lang.org/std/primitive.pointer.html#method.read_unaligned)
+
+[^38]: [https://doc.rust-lang.org/std/primitive.pointer.html#method.write_unaligned](https://doc.rust-lang.org/std/primitive.pointer.html#method.write_unaligned)
+
+[^39]: These are effectively memcpys: they will behave as if they read each byte individually with no respect for alignment.
+
+[^40]: [https://doc.rust-lang.org/std/primitive.pointer.html#method.copy_to](https://doc.rust-lang.org/std/primitive.pointer.html#method.copy_to)
+
+[^41]: [https://doc.rust-lang.org/std/primitive.pointer.html#method.copy_to_nonoverlapping](https://doc.rust-lang.org/std/primitive.pointer.html#method.copy_to_nonoverlapping)
```

Comment:
40 and 41 are also messed up where they are referenced in the text and don't appear at the end in the visualization on github. Maybe it's because they are only links with no text?

Reply:
I'm pretty confused why this is occurring if github markdown doesn't support footnotes...

Reply:
There are more (such as 56 and 57 below) but I'm going to stop listing them because 1) I'm definitely going to miss some anyways and 2) I'm not sure if it's a bug you care about.